### PR TITLE
Hash-Join materialize input 

### PR DESCRIPTION
**Pull request by marcelja**
This removes the vector `materialized_chunk` and calculates the histograms in the iterable. Thus, there is one less loop.

screen shot 2018-08-07 at 11 22 20

### CODE REVIEW

File: src/lib/operators/join_hash.cpp [hunk 1]
```diff
@@ -217,69 +217,43 @@ std::shared_ptr<Partition<T>> materialize_input(const std::shared_ptr<const Tabl
...
-
-          const Hash radix = (output[row_id].partition_hash >> (32 - radix_bits * (pass + 1))) & mask;
-          histogram[radix]++;
-
-          row_id++;
-        }
-      }
+          // reference_column_offset is only used for ReferenceColumns
+          if constexpr (std::is_same<std::decay<decltype(typed_column)>, ReferenceColumn>::value) {
+            reference_column_offset++;
```
Comment (mrks) [inline src/lib/operators/join_hash.cpp]: I would have assumed that the compiler finds this itself. It's better to be explicit anyway, but did you test it?

Comment (marcelja) [reply src/lib/operators/join_hash.cpp]: That's true. No, I didn't test it.

Maybe it would be easier to read without the if?

Comment (mrks) [reply src/lib/operators/join_hash.cpp]: All fine, I was just curious.

Comment (marcelja) [reply src/lib/operators/join_hash.cpp]: Ok 😄

**Comment by Bouncner** [comment]
Hey @marcelja , what TPC-H scale was that?

**Comment by marcelja** [comment]
Hey, that was scale factor 1.

**Pull request merged**